### DISCUSSION TITLE
GUI not activating - PyQt5 and PySide2 Errors

### DISCUSSION THREAD

**Discussion by a8cone** [talmolab/sleap#1615] [Help!]
Hi again,

I updated sleap to 1.3.3, but when I try to activate the GUI, I get an error. I've provided the error below. Does anyone have any recommendations?

![image](https://github.com/talmolab/sleap/assets/60786631/e6c1aa76-6340-4434-b754-460a9e0de751)

**Here is the sleap-diagnostic:**

![image](https://github.com/talmolab/sleap/assets/60786631/7245a2cf-4291-4666-a2f6-f27d2fa50c58)
![image](https://github.com/talmolab/sleap/assets/60786631/5c8f6ae1-fe6b-475c-9467-0955e7ba2638)
![image](https://github.com/talmolab/sleap/assets/60786631/5321ec8f-491f-4dab-ab76-0eedc7400359)
![image](https://github.com/talmolab/sleap/assets/60786631/141bd205-fcd3-46cd-959a-0cc5e47975c0)
![image](https://github.com/talmolab/sleap/assets/60786631/c5783ee3-fea8-4087-8361-b8502e673913)
![image](https://github.com/talmolab/sleap/assets/60786631/223290bf-cc31-4b95-9bcc-ca00cc8044b9)

Aaron

**Comment by talmo** [comment]
Hi @a8cone,

Do you mind trying this in a fresh environment? Basically:

```
conda env remove -n sleap
```

Then:

```
mamba create -y -n sleap -c conda-forge -c nvidia -c sleap -c anaconda sleap=1.3.3
```

Let us know if you're still having issues after deleting the environment and then reinstalling.

Thanks!

Talmo

**Reply by a8cone** [reply]
Hi @talmo,

I tried your recommendation but I am still getting the error. 

I do not have mamba installed. 

So I did this:

`conda env remove -n sleap`

`conda create -y -n sleap -c conda-forge -c nvidia -c sleap -c anaconda sleap=1.3.3`

![image](https://github.com/talmolab/sleap/assets/60786631/62f86a41-093d-4a36-b72c-1254c9596138)


Any other recommendations?

Thank you in advance.

Aaron

**Reply by talmo** [reply]
Well that's strange! Ok, let's investigate.

The issue is that your environment seems to want to install `PyQt5==5.12` (plus whatever `PyQt5_sip==4.19.19` is).

PyQt5 is an alternative to PySide2, which works on some platforms where the other doesn't. From the error you're getting, that version of PyQt5 isn't compatible, but this shouldn't be an issue since we use `qtpy` which provides a compatibility layer...

Stranger still, your logs indicate that you _do_ have pyside2 installed since it's importable (second screenshot), but your package list doesn't list it (fifth screenshot)!

I think this might create more issues, but one thing to try to uninstall pyqt5:

```
pip uninstall PyQt5
```

Then, you could force reinstall PySide2:

```
pip install --ignore-installed PySide2==5.15.6
```

If none of that works, for now you can try [installing from source](https://sleap.ai/installation.html#conda-from-source) as an alternative while we sort this out.

Let us know how the above steps go!

Cheers,

Talmo

**Reply by a8cone** [reply]
@talmo,

So I installed sleap from source but I am still getting the error. 

I used:

`conda env remove -n sleap`

`git clone https://github.com/talmolab/sleap && cd sleap`

`conda env create -f environment.yml -n sleap`

![image](https://github.com/talmolab/sleap/assets/60786631/4f3af072-cbac-42d0-a510-ff35bae17c60)
![image](https://github.com/talmolab/sleap/assets/60786631/d2d46200-3c6a-413b-b529-edd54ae3f016)


Thank you for your help in advance.

Aaron

**Reply by talmo** [reply]
Hmm, looks like you tried **installing** PyQt5 -- do you mind trying PySide2?

```
pip uninstall PyQt5
pip install --ignore-installed PySide2==5.15.6
```

**Reply by a8cone** [reply]
@talmo,

I tried what you suggested. But I got an error with PySide2==5.15.6. So I just defaulted to version that would allow install. Please see images below.

![image](https://github.com/talmolab/sleap/assets/60786631/6afefbb3-486b-41dd-88c4-d83dbf379337)
![image](https://github.com/talmolab/sleap/assets/60786631/6a15f186-2341-4079-b6d7-0935b925b7d6)


Aaron

**Reply by talmo** [reply]
Ok, hmmmmm.... This is very odd. We're totally unable to reproduce this on our end on Windows...

More things that could be the cause:

1. You're on Windows 10.0.19041 (20H1), which is a few years old. You could try updating it, which might pull in some dependencies that are causing issues with the GUI packages.
2. You're using Anaconda instead of Miniconda or Mambaforge. I don't think this is an issue but it's one of the things that's different.

Sorry!! Alternatively, you mentioned that this happened after updating -- do you get the same issue with previous versions of SLEAP? I'd try 1.3.0 or 1.2 for troubleshooting.

**Reply by a8cone** [reply]
@talmo,

Interestingly I am only able to activate the sleap gui by sleap_env. I used [(https://sleap.ai/guides/installation.html )](url) as a reference. 

I used:

`conda create -n sleap_env -c defaults -c sleap sleap=1.0 python=3.6 -y`
`conda activate sleap_env`
`sleap-label`

I also had to install tensorflow-estimator-2.1.0 and attrs==19.3 for it to work.

But with this version, the issue with the prediction training and inference occurred. [https://github.com/talmolab/sleap/discussions/1614](url)



Any suggestions? 

Thanks for your help in advance. 

Aaron

**Reply by talmo** [reply]
Hi @a8cone,

It looks like you've found one of our older installation pages for a _very_ old version of SLEAP (v1.0) from several years ago.

Do you mind trying [v1.2.9](https://github.com/talmolab/sleap/releases/tag/v1.2.9) or [v1.3.0](https://github.com/talmolab/sleap/releases/tag/v1.3.0)?

Additionally, have you tried to update your Windows? (Just go to Windows Update to update to the latest Windows 10 -- you don't have to upgrade to Windows 11.)

**Reply by a8cone** [reply]
@talmo,

I tried installing v1.2.9 and v1.3.0. I am still getting the error regarding PyQt5. 

I can maybe get it to work on a different machine, but the current machine has a GPU in it to utilize. 

The machine is up to date as well.
![image](https://github.com/talmolab/sleap/assets/60786631/40f03713-de16-4c18-bff1-0e6678fb6ea3)

Thanks for the help.

Aaron

**Reply by talmo** [reply]
Hi @a8cone,

Right, so as the red asterisk message says, it might be that your computer is configured to prevent major version updates, including to newer Windows 10 releases. Maybe you can ask your IT or whoever set up the computer to help? (It's a settings in the Group Policy applet.)

Again, I'm not sure if this is the issue, but I can't think of what else might be causing this problem.

If you can't update that computer, perhaps there's another one that you could test out the installation on?

Let us know if you have any questions!

Talmo
